api4.php

```diff
@@ -18,14 +18,22 @@
  * @return \Civi\Api4\Generic\Result
  */
 function civicrm_api4($entity, $action, $params = []) {
-  $params['version'] = 4;
   // For custom pseudo-entities
   if (strpos($entity, 'Custom_') === 0) {
-    $params['customGroup'] = substr($entity, 7);
-    $entity = 'CustomValue';
+    $apiCall = Civi\Api4\CustomValue::$action(substr($entity, 7));
   }
-  $request = \Civi\API\Request::create($entity, $action, $params);
-  return \Civi::service('civi_api_kernel')->runRequest($request);
+  else {
+    $callable = ["Civi\\Api4\\$entity", $action];
+    if (!is_callable($callable)) {
+      throw new Exception\NotImplementedException("API ($entity, $action) does not exist (join the API team and implement it!)");
+    }
+    $apiCall = call_user_func($callable);
+  }
+  foreach ($params as $name => $param) {
+    $setter = 'set' . ucfirst($name);
+    $apiCall->$setter($param);
```

Comment:
In `cv api4`, it calls `civicrm_api4(...)` and includes `$params['version'] === 4`. This now throws an exception `Cannot modify api version` (because it's blacklisted via `AbstractAction::setVersion()`).

I'll update `cv api4` to... just not do that. The version is implicit in calling `civicrm_api4()`. (The `version` is only necessary when going through `civicrm_api()` - and that still seems to work OK.) However, I mention in case that's an unintended side-effect.

Reply:
That seems unnecessarily strict. I think I'll modify that error to only be thrown if you try to set it to a value other than `4`.

Reply:
@totten: updated.